editor/directory_create_dialog.cpp

```diff
@@ -0,0 +1,172 @@
+#include "editor/editor_node.h"
+#include "editor/editor_scale.h"
+#include "scene/gui/box_container.h"
+#include "scene/gui/label.h"
+#include "scene/gui/line_edit.h"
+#include "scene/gui/panel_container.h"
+
+static String sanitize_input(const String &p_path) {
+	String path = p_path.strip_edges();
+	if (path.ends_with("/")) {
+		path = path.left(path.length() - 1);
+	}
+	return path;
+}
+
+String DirectoryCreateDialog::_validate_path(const String &p_path) const {
+	if (p_path.is_empty()) {
+		return TTR("Folder name cannot be empty.");
+	}
+
+	for (const String &part : p_path.split("/")) {
+		if (part.is_empty()) {
+			return TTR("Folder name cannot be empty.");
+		}
+		if (p_path.contains("\\") || p_path.contains(":") || p_path.contains("*") ||
+				p_path.contains("|") || p_path.contains(">") || p_path.ends_with(".") || p_path.ends_with(" ")) {
+			return TTR("Folder name contains invalid characters.");
+		}
+	}
+
+	Ref<DirAccess> da = DirAccess::create(DirAccess::ACCESS_RESOURCES);
+	da->change_dir(base_dir);
+	if (da->file_exists(p_path)) {
+		return TTR("File with that name already exists.");
+	}
+	if (da->dir_exists(p_path)) {
+		return TTR("Folder with that name already exists.");
+	}
+
+	return String();
+}
+
+void DirectoryCreateDialog::_on_dir_path_changed(const String &p_text) {
+	const String path = sanitize_input(p_text);
+	const String error = _validate_path(path);
+
+	if (error.is_empty()) {
+		status_label->add_theme_color_override("font_color", get_theme_color(SNAME("success_color"), SNAME("Editor")));
+
+		if (path.contains("/")) {
+			status_label->set_text(TTR("Using slashes in folder names will create subfolders recursively."));
+		} else {
+			status_label->set_text(TTR("Folder name is valid."));
+		}
+	} else {
+		status_label->add_theme_color_override("font_color", get_theme_color(SNAME("error_color"), SNAME("Editor")));
+		status_label->set_text(error);
+	}
+
+	get_ok_button()->set_disabled(!error.is_empty());
+}
+
+void DirectoryCreateDialog::ok_pressed() {
+	const String path = sanitize_input(dir_path->get_text());
+
+	// The OK button should be disabled if the path is invalid, but just in case.
+	const String error = _validate_path(path);
+	ERR_FAIL_COND_MSG(!error.is_empty(), error);
+
+	Error err;
+	Ref<DirAccess> da = DirAccess::create(DirAccess::ACCESS_RESOURCES);
+
+	err = da->change_dir(base_dir);
+	ERR_FAIL_COND_MSG(err != OK, "Cannot open directory '" + base_dir + "'.");
+
+	print_verbose("Making folder " + path + " in " + base_dir);
+	err = da->make_dir_recursive(path);
+
+	if (err == OK) {
+		emit_signal(SNAME("dir_created"));
+	} else {
+		EditorNode::get_singleton()->show_warning(TTR("Could not create folder."));
+	}
+	hide();
+}
+
+void DirectoryCreateDialog::_post_popup() {
+	dir_path->grab_focus();
+}
+
+void DirectoryCreateDialog::config(const String &p_base_dir) {
+	base_dir = p_base_dir;
+	label->set_text(vformat(TTR("Create new folder in %s:"), base_dir));
+	dir_path->set_text("new folder");
```

Comment:
Shouldn't this be translated?

Reply:
The name was not translated in the old dialog.

I'm not sure if it is fixed now, but Android export had trouble loading non-ascii paths / filenames.